From 582d885074c86d8c7c56c390dad7113456552aeb Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 26 Jul 2012 09:54:13 +0800
Subject: ENGR00218412-2 OV5642:Power down after checking dev id

This patch powers down camera after checking camera device
id to save power.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
(cherry picked from commit eb280a7182cd8c71d698b57a720447f9d9b1174a)
---
 drivers/media/video/mxc/capture/ov5642.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/video/mxc/capture/ov5642.c b/drivers/media/video/mxc/capture/ov5642.c
index f3a82db..245a92c 100644
--- a/drivers/media/video/mxc/capture/ov5642.c
+++ b/drivers/media/video/mxc/capture/ov5642.c
@@ -4011,6 +4011,9 @@ static int ov5642_probe(struct i2c_client *client,
 		goto err4;
 	}
 
+	if (plat_data->pwdn)
+		plat_data->pwdn(1);
+
 	camera_plat = plat_data;
 
 	ov5642_int_device.priv = &ov5642_data;
-- 
1.7.9.5

